tests/integration/sync/sync_integ_base.py

```diff
@@ -1,19 +1,141 @@
 import os
+
+import logging
 import json
-from unittest import TestCase
+import shutil
+import time
+import uuid
+from pathlib import Path
+
+import boto3
+from botocore.exceptions import ClientError
+from botocore.config import Config
+
+from samcli.lib.bootstrap.bootstrap import SAM_CLI_STACK_NAME
+from tests.integration.buildcmd.build_integ_base import BuildIntegBase
+from tests.integration.package.package_integ_base import PackageIntegBase
 
+CFN_SLEEP = 3
+CFN_PYTHON_VERSION_SUFFIX = os.environ.get("PYTHON_VERSION", "0.0.0").replace(".", "-")
+RETRY_ATTEMPTS = 20
+RETRY_WAIT = 1
 
-class SyncIntegBase(TestCase):
+LOG = logging.getLogger(__name__)
+
+
+class SyncIntegBase(BuildIntegBase, PackageIntegBase):
     @classmethod
     def setUpClass(cls):
-        pass
+        PackageIntegBase.setUpClass()
+
+        cls.test_data_path = Path(__file__).resolve().parents[1].joinpath("testdata", "sync")
 
     def setUp(self):
+        self.cfn_client = boto3.client("cloudformation")
+        self.ecr_client = boto3.client("ecr")
+        self.lambda_client = boto3.client("lambda")
+        self.api_client = boto3.client("apigateway")
+        self.sfn_client = boto3.client("stepfunctions")
+        self.sns_arn = os.environ.get("AWS_SNS")
+        self.stacks = []
+        self.s3_prefix = uuid.uuid4().hex
         super().setUp()
 
     def tearDown(self):
+        shutil.rmtree(os.path.join(os.getcwd(), ".aws-sam"), ignore_errors=True)
+        for stack in self.stacks:
+            # because of the termination protection, do not delete aws-sam-cli-managed-default stack
+            stack_name = stack["name"]
+            if stack_name != SAM_CLI_STACK_NAME:
+                region = stack.get("region")
+                cfn_client = (
+                    self.cfn_client if not region else boto3.client("cloudformation", config=Config(region_name=region))
+                )
+                ecr_client = self.ecr_client if not region else boto3.client("ecr", config=Config(region_name=region))
+                self._delete_companion_stack(cfn_client, ecr_client, self._stack_name_to_companion_stack(stack_name))
+                cfn_client.delete_stack(StackName=stack_name)
         super().tearDown()
 
+    def _get_stacks(self, stack_name):
```

Comment:
Seems like function name doesn't match what we are returning. It says `_get_stacks` but it returns `physical_ids`. Can we name the function better?